From 2ab3776a77375ab3c36500e6a1a416e30a679118 Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@foss.st.com>
Date: Wed, 13 Mar 2024 11:53:10 +0100
Subject: [PATCH] arm64: dts: st: change mdf sitf frequency on stm32mp257f-dk

The mdf decimation ratio can be expressed as Fsitf / Fs where
Fsitf is the frequency of the serial interface wired to digital
microphones, and Fs the PCM sampling rate.
With Fsitf = 2048 MHz, the 48kHz sampling rate is not supported
when the reshape filter is active.
Change Fsitf to 1536MHz support 8,16, 32 and 48kHz rates with reshape
filter.
With filter order 5, the decimation ratio is out of range for 8kHz rate.
So, reduce also filter order to 4.

Signed-off-by: Olivier Moysan <olivier.moysan@foss.st.com>
Change-Id: I39b1ef613c2a24fe630f6018bd3faa2ee10b8c4e
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/368952
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-dk.dts | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
index 530784679c75..13f6bdd8d3f6 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
@@ -484,7 +484,7 @@ &mdf1 {
 	pinctrl-1 = <&mdf_cck0_sleep_pins_a>;
 	#clock-cells = <1>;
 	clock-output-names = "cck0";
-	clock-frequency = <2048000>;
+	clock-frequency = <1536000>;
 	status = "okay";
 
 	sitf6: sitf@380 {
@@ -497,7 +497,7 @@ sitf6: sitf@380 {
 	};
 
 	filter0: filter@84  {
-		st,cic-mode = <5>;
+		st,cic-mode = <4>;
 		st,sitf = <&sitf6 0>;
 		st,hpf-filter-cutoff-bp = <625>;
 		status = "okay";
@@ -518,7 +518,7 @@ mdf_endpoint0: endpoint {
 	};
 
 	filter1: filter@104  {
-		st,cic-mode = <5>;
+		st,cic-mode = <4>;
 		st,sitf = <&sitf6 1>;
 		st,hpf-filter-cutoff-bp = <625>;
 		status = "okay";
-- 
2.39.5

